src/Microsoft.CSharp/tests/CSharpArgumentInfoTests.cs

```diff
@@ -1,18 +1,33 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace Microsoft.CSharp.RuntimeBinder.Tests
 {
     public class CSharpArgumentInfoTests
     {
-        [Fact]
-        public void Create_ResultNotNull()
+        private static readonly IEnumerable<CSharpArgumentInfoFlags> AllPossibleFlags =
+            Enumerable.Range(0, ((int[])Enum.GetValues(typeof(CSharpArgumentInfoFlags))).Max() * 2)
+                .Select(i => (CSharpArgumentInfoFlags)i);
+
+        private static readonly string[] Names =
+        {
+            "arg", "ARG", "Arg", "Argument name that isn’t a valid C♯ name 👿🤢",
+            "horrid name with" + (char)0xD800 + "a half surrogate", "new", "break", null
+        };
+
+        private static IEnumerable<object[]> FlagsAndNames() =>
+            AllPossibleFlags.Select((f, i) => new object[] {false, Names[i % Names.Length]});
```

Comment:
Yes. Another example of how it's okay to commit last thing before going to bed, but not to push the commit.